weather_mv/loader_pipeline/ee_util.py

```diff
@@ -0,0 +1,457 @@
+    projection: str  # The gdal string defining the projection.
+    transform: t.Tuple[float, float, float, float, float, float]  # From gdal.
+    metadata: t.Dict[str, t.Union[str, float, int]]
+    data: t.List[xr.Dataset]  # The list of datasets
+
+
+@dataclasses.dataclass
+class ToEarthEngine(ToDataSink):
+    """Loads weather data into Google Earth Engine.
+
+    TODO(alxr): Document...
+    """
+    example_uri: str
+    xarray_open_dataset_kwargs: t.Dict
+    disable_in_memory_copy: bool
+    disable_grib_schema_normalization: bool
+    skip_region_validation: bool
+    ee_asset: str
+    tiff_location: str
+    use_personal_account: bool
+
+    @classmethod
+    def add_parser_arguments(cls, subparser: argparse.ArgumentParser):
+        subparser.add_argument('--xarray_open_dataset_kwargs', type=json.loads, default='{}',
+                               help='Keyword-args to pass into `xarray.open_dataset()` in the form of a JSON string.')
+        subparser.add_argument('--disable_in_memory_copy', action='store_true', default=False,
+                               help='To disable in-memory copying of dataset. Default: False')
+        subparser.add_argument('--disable_grib_schema_normalization', action='store_true', default=False,
+                               help='To disable merge of grib datasets. Default: False')
+        subparser.add_argument('--ee_asset', type=str, default=None,
+                               help='The asset folder path in earth engine project where the tiff image files'
+                               ' will be pushed.')
+        subparser.add_argument('-s', '--skip-region-validation', action='store_true', default=False,
+                               help='Skip validation of regions for data migration. Default: off')
+        subparser.add_argument('--tiff_location', type=str, default=None,
+                               help='The GCS location where the tiff files will be pushed.')
+        subparser.add_argument('-u', '--use_personal_account', action='store_true', default=False,
+                               help='To use personal account for earth engine authentication.')
+
+    @classmethod
+    def validate_arguments(cls, known_args: argparse.Namespace, pipeline_args: t.List[str]) -> None:
+        pipeline_options = PipelineOptions(pipeline_args)
+        pipeline_options_dict = pipeline_options.get_all_options()
+
+        if not known_args.tiff_location:
+            raise RuntimeError("'--tiff_location' is required for ee ingestion.")
+        if not known_args.ee_asset:
+            raise RuntimeError("'--ee_asset' is required for ee ingestion.")
+        if not re.match("^projects/.+/assets.*", known_args.ee_asset):
+            raise RuntimeError("'--ee_asset' is required to be in format: projects/+/assets/*.")
+
+        # Check that Cloud resource regions are consistent.
+        if not (known_args.dry_run or known_args.skip_region_validation):
```

Comment:
Is this necessary for the EE API to work? Mainly, I'm curious.

Reply:
Yes, it's the directory path where we want to push the tiff image into the earth engine.
If not provided, it throws an exception saying the name is invalid.

Assets under the top-level folders "users" or "projects" have names of the format "projects/earthengine-legacy/assets/**" (e.g. "projects/earthengine-legacy/assets/users/foo/bar"). Public assets have the format "projects/earthengine-public/assets/**" (e.g. "projects/earthengine-public/assets/LANDSAT"). Assets under a custom cloud project have the format "projects/*/assets/**" (e.g. "projects/my-gcp-project/assets/my/foo/bar").